### PR TITLE
feat(tree-sitter): :sparkles: add go template support

### PR DESCRIPTION
**Pull request by dannylongeuay**
Adds Go Template highlighting support.

image

### CODE REVIEW

File: runtime/queries/gotmpl/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,75 @@
+; Identifiers
+
+[
+  (field)
+  (field_identifier)
+] @property
```
Comment (the-mikedavis) [inline runtime/queries/gotmpl/highlights.scm:6]:
```suggestion
] @variable.other.member
```

File: runtime/queries/gotmpl/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  (rune_literal)
+] @string
+
+(escape_sequence) @string.special
+
+[
+  (int_literal)
+  (float_literal)
+  (imaginary_literal)
+] @number
```
Comment (archseer) [inline runtime/queries/gotmpl/highlights.scm:66]: This should also use the proper scopes

File: runtime/queries/gotmpl/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+[
+  (true)
+  (false)
+  (nil)
+] @constant.builtin
```
Comment (archseer) [inline runtime/queries/gotmpl/highlights.scm:72]: Split the bools out and tag them as constant.builtin.boolean

File: runtime/queries/gotmpl/highlights.scm (hunk 1 above)
Comment (dannylongeuay) [reply runtime/queries/gotmpl/highlights.scm]: fixed

File: runtime/queries/gotmpl/highlights.scm (hunk 2 above)
Comment (dannylongeuay) [reply runtime/queries/gotmpl/highlights.scm]: fixed

File: runtime/queries/gotmpl/highlights.scm (hunk 3 above)
Comment (dannylongeuay) [reply runtime/queries/gotmpl/highlights.scm]: fixed

**Comment by dannylongeuay** [comment]
Thanks for the feedback. I have addressed the requested changes and comments.

**Review by archseer — APPROVED** [review]
Thanks!

**Pull request merged**